X11R6/xc/doc/man/Xi/XDevBell.man
.\"
.\" Copyright ([\d,\s]*) by Hewlett-Packard Company, Ardent Computer, 
.\" 
.\" Permission to use, copy, modify, distribute, and sell this documentation 
.\" for any purpose and without fee is hereby granted, provided that the above
.\" copyright notice and this permission notice appear in all copies.
.\" Ardent, and Hewlett-Packard make no representations about the 
.\" suitability for any purpose of the information in this document.  It is 
.\" provided \`\`as is'' without express or implied warranty.
.\" 
.\" $XConsortium: XDevBell.man,v 1.6 94/06/04 17:32:23 rws Exp $
.ds xL Programming With Xlib
.TH XDeviceBell 3X11 "Release 6" "X Version 11" "X FUNCTIONS"
.SH NAME
XDeviceBell \- ring a bell on a device supported through the input extension
.SH SYNTAX
Status XDeviceBell\^(\^\fIdisplay\fP, \fIdevice\fP\^, \fIfeedbackclass\fP\^, \fIfeedbackid\fP\^, \fIpercent\fP\^)
.br
      Display *\fIdisplay\fP\^;
.br
      XDevice *\fIdevice\fP\^; 
.br
      XID *\fIfeedbackclass\fP\^; 
.br
      XID *\fIfeedbackid\fP\^; 
.br
      int *\fIpercent\fP\^; 
.br
.SH ARGUMENTS
.TP 12
.I display
Specifies the connection to the X server.
.TP 12
.I device
Specifies the device with which the bell is associated.
.TP 12
.I feedbackclass
Specifies the class of the feedback with which the bell is associated.
.TP 12
.I feedbackid
Specifies the id of the feedback with which the bell is associated.
.TP 12
.I percent
Specifies the volume in the range -100 to 100 at which the bell should be rung.
.SH DESCRIPTION
The \fIXDeviceBell\fP request causes the server to ring a bell on the 
specified feedback of the specified device, if possible.  
The specified volume is relative to the base volume for the bell.
If an invalid device is specified,
a \fIBadDevice\fP error will be returned.  The feedbackclass and feedbackid
parameters contain values returned by an \fIXGetFeedbackControl\fP request
and uniquely identify the bell to ring.  If a feedbackclass is specified that
does not support a bell, or if a nonexistent feedbackid is specified,
or a percent value is specified that is not in the range -100 to 100,
a \fIBadValue\fP error will be returned.
.LP
The volume at which the bell is rung when the percent argument is 
nonnegative is:
.IP
base \- [(base * percent) / 100] + percent
.LP
The volume at which the bell rings
when the percent argument is negative is:
.IP
base + [(base * percent) / 100]
.LP
To change the base volume of the bell, use \fIXChangeFeedbackControl\fP.
.LP
\fIXDeviceBell\fP can generate a \fIBadDevice\fP or a \fIBadValue\fP
error.
.SH DIAGNOSTICS
.TP 12
\fIBadDevice\fP
An invalid device was specified.  The specified device does not exist, or has 
not been opened by this client via \fIXOpenInputDevice\fP.
.TP 12
\fIBadValue\fP
An invalid feedbackclass, feedbackid, or percent value was specified.
.SH "SEE ALSO"
XChangeFeedbackControl(3X),
XBell(3X)
.br
\fI\*(xL\fP


X11R6/xc/lib/Xi/XDevBell.c
/* $XConsortium: XDevBell.c,v 1.5 94/04/17 20:17:59 rws Exp $ */

/************************************************************

Copyright (c) 1989  X Consortium

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
X CONSORTIUM BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

Except as contained in this notice, the name of the X Consortium shall not be
used in advertising or otherwise to promote the sale, use or other dealings
in this Software without prior written authorization from the X Consortium.

Copyright (c) 1989 by Hewlett-Packard Company, Palo Alto, California.

			All Rights Reserved

Permission to use, copy, modify, and distribute this software and its
documentation for any purpose and without fee is hereby granted,
provided that the above copyright notice appear in all copies and that
both that copyright notice and this permission notice appear in
supporting documentation, and that the name of Hewlett-Packard not be
used in advertising or publicity pertaining to distribution of the
software without specific, written prior permission.

HEWLETT-PACKARD DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING
ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT SHALL
HEWLETT-PACKARD BE LIABLE FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR
ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
SOFTWARE.

********************************************************/

/***********************************************************************
 *
 * XDeviceBell - Ring a bell on an extension device.
 *
 */

#include "XI.h"
#include "XIproto.h"
#include "Xlibint.h"
#include "XInput.h"
#include "extutil.h"

int
XDeviceBell (dpy, dev, feedbackclass, feedbackid, percent)
    register Display 	*dpy;
    XDevice		*dev;
    XID			feedbackclass, feedbackid;
    int			percent;
    {       
    xDeviceBellReq 	*req;
    XExtDisplayInfo *info = (XExtDisplayInfo *) XInput_find_display (dpy);

    LockDisplay (dpy);
    if (_XiCheckExtInit(dpy, XInput_Add_XDeviceBell) == -1)
	return (NoSuchExtension);

    GetReq(DeviceBell,req);		
    req->reqType = info->codes->major_opcode;
    req->ReqType = X_DeviceBell;
    req->deviceid = dev->device_id;
    req->feedbackclass = feedbackclass;
    req->feedbackid = feedbackid;
    req->percent = percent;

    UnlockDisplay(dpy);
    SyncHandle();
    return (Success);
    }

